### PR TITLE
improve/add cleanup in all specs in the 'flow' package

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy [hunk 1]
```diff
@@ -85,8 +86,8 @@ class FlowValidationNegativeSpec extends HealthCheckSpecification {
             assert nonAffectedSwitches.every { sw -> northbound.validateSwitchRules(sw).excessRules.size() == 0 }
         }
 
-        and: "Delete the flows"
-        [flowToBreak.id, intactFlow.id].each { flowHelper.deleteFlow(it) }
+        cleanup: "Delete the flows"
+        [flowToBreak?.id, intactFlow?.id].each { flowHelper.deleteFlow(it) }
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy:90]: same

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeV2Spec.groovy [hunk 2]
```diff
@@ -84,7 +84,7 @@ class FlowValidationNegativeV2Spec extends HealthCheckSpecification {
         }
 
         cleanup: "Delete the flows"
-        [flowToBreak.flowId, intactFlow.flowId].each { flowHelperV2.deleteFlow(it) }
+        [flowToBreak?.flowId, intactFlow?.flowId].each { flowHelperV2.deleteFlow(it) }
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeV2Spec.groovy:87]: ^

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowV2Spec.groovy [hunk 3]
```diff
@@ -159,9 +160,18 @@ class PinnedFlowV2Spec extends HealthCheckSpecification {
...
-        and: "Cleanup: revert system to original state"
-        flowHelperV2.deleteFlow(flow.flowId)
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        if (islsToBreak && !islsAreUp) {
+            islsToBreak.each { islToRestore ->
+                antiflap.portUp(islToRestore.srcSwitch.dpId, islToRestore.srcPort)
+                Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+                    assert islUtils.getIslInfo(islToRestore).get().state == IslChangeType.DISCOVERED
+                }
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowV2Spec.groovy:172]: Not the most efficient way to restore in cleanup. In test these steps were intentionally used to restore isls one by one, but in cleanup it'll be better to request all portUps first and the do the waiting for isls.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy]: fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeV2Spec.groovy (hunk 2 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeV2Spec.groovy]: fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowV2Spec.groovy (hunk 3 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowV2Spec.groovy]: nice catch, fixed

**Review by rtretyak — APPROVED** [review]

**Pull request merged**